### PR TITLE
Refactor async code into actions

### LINKED ISSUE #17: Refactor async code into actions
- LoginScreen
- GymDetailsScreen (useEffect)
- UpdateProfileScreen
- RegisterScreen

### PR DESCRIPTION
**Pull request by FirescuOvidiu**
[Issue:17](https://github.com/FirescuOvidiu/Gym-Application/issues/17)

### CODE REVIEW

File: Frontend/src/redux/thunks/userThunks.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    try {
+      let user = await axios.get('http://192.168.100.2:3000/api/user', {
+        headers: {
+          authorization: token,
+        },
+      });
+
+      if (user.data.user) {
+        delete user.data.user.__v;
+        delete user.data.user.password;
```
Comment (AdiCutitoiu) [inline Frontend/src/redux/thunks/userThunks.js]: The password shouldn't be contained in the response.

File: Frontend/src/redux/thunks/userThunks.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    const token = await AsyncStorage.getItem('accessToken');
+
+    try {
+      let user = await axios.get('http://192.168.100.2:3000/api/user', {
+        headers: {
+          authorization: token,
+        },
+      });
+
+      if (user.data.user) {
```
Comment (AdiCutitoiu) [inline Frontend/src/redux/thunks/userThunks.js]: Is there any scenario where user.data.user is null? If so, the lines below which are not included in this if statement will crash the app.

Comment (FirescuOvidiu) [reply Frontend/src/redux/thunks/userThunks.js]: There is no scenario where user.data.user is null, removed the if.

File: Frontend/src/redux/thunks/userThunks.js (hunk 1 above)
Comment (FirescuOvidiu) [reply Frontend/src/redux/thunks/userThunks.js]: It wasn't contained in the response, removed unnecessary code.

File: Frontend/src/redux/thunks/gymThunks.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+const axios = require('axios').default;
+
+import {addGym} from '../actions/gymActions';
+
+export const saveGym = () => {
+  return async (dispatch) => {
+    const token = await AsyncStorage.getItem('accessToken');
+
+    try {
+      let gym = await axios.get('http://192.168.100.2:3000/api/gym', {
```
Comment (RaduAlbastroiu) [inline Frontend/src/redux/thunks/gymThunks.js:12]: You should use axios instance to set base url.
But we'll talk about this next time.
[NO ACTION REQUIRED]

File: Frontend/src/redux/thunks/gymThunks.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        headers: {
+          authorization: token,
+        },
+      });
+
+      if (gym.data.gym) {
+        delete gym.data.gym[0].__v;
+      }
```
Comment (RaduAlbastroiu) [inline Frontend/src/redux/thunks/gymThunks.js]: It's not necessary to delete the version of the document.

File: Frontend/src/redux/thunks/gymThunks.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+      dispatch(addGym(gym.data.gym[0]));
```
Comment (RaduAlbastroiu) [inline Frontend/src/redux/thunks/gymThunks.js]: Why gym[0]? Why is this an array?
Are there multiple gyms? (if not it should be just gym.data )

File: Frontend/src/redux/thunks/userThunks.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    try {
+      let user = await axios.get('http://192.168.100.2:3000/api/user', {
+        headers: {
+          authorization: token,
+        },
+      });
+
+      user = user.data.user;
+      user.birthday = user.birthday.substring(0, 10);
+      user.date = user.date.substring(0, 10);
```
Comment (RaduAlbastroiu) [inline Frontend/src/redux/thunks/userThunks.js:48]: Why substring 0, 10?
Why not use same data format on backend as frontend?
Or do date formating on ui.

File: Frontend/src/redux/thunks/userThunks.js [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+      dispatch(addUser(user));
+    } catch (error) {
+      alert(error.response.data.message);
+    }
+  };
+};
+
+export const _updateUser = ({
```
Comment (RaduAlbastroiu) [inline Frontend/src/redux/thunks/userThunks.js:56]: _ does it have a purpose or is it a mistake?

File: Frontend/src/redux/thunks/userThunks.js [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+  userReducer,
+  user,
+  setUserModified,
```
Comment (RaduAlbastroiu) [inline Frontend/src/redux/thunks/userThunks.js:59]: We shall discuss this in our next meeting

File: Frontend/src/screens/gymDetailsScreen.js [hunk 9]
```diff
@@ -14,55 +14,16 @@ import AsyncStorage from '@react-native-community/async-storage';
...
-        }
-        dispatch(addGym(gym.data.gym[0]));
-      } catch (error) {
-        alert(error.response.data.message);
-      }
-    };
-    saveUser();
-    saveGym();
+    dispatch(saveUser());
+    dispatch(saveGym());
```
Comment (RaduAlbastroiu) [inline Frontend/src/screens/gymDetailsScreen.js:26]: beautiful isn't it?
[NO ACTION REQUIRED]

**Review by RaduAlbastroiu — CHANGES_REQUESTED** [review]

File: Frontend/src/redux/thunks/gymThunks.js (hunk 5 above)
Comment (FirescuOvidiu) [reply Frontend/src/redux/thunks/gymThunks.js]: There is a single gym. Resolved the issue.

File: Frontend/src/redux/thunks/userThunks.js (hunk 6 above)
Comment (FirescuOvidiu) [reply Frontend/src/redux/thunks/userThunks.js:48]: The data is saved in the database with the type Date and it adds at the end of the date the hours, minutes, seconds which are irrelevant.

File: Frontend/src/redux/thunks/userThunks.js (hunk 7 above)
Comment (FirescuOvidiu) [reply Frontend/src/redux/thunks/userThunks.js:56]: I have another function with the name updateUser in userActions which is used in the same file as this one.

File: Backend/api/gym/controller.js [hunk 10]
```diff
@@ -7,13 +7,15 @@ const getGym = async (req, res, next) => {
...
-      gym = await Gym.find();
+      gym = await Gym.findOne();
     }
 
     if (!gym) {
-      return next({ message: "The user was not found." });
+      return next({ message: "The gym was not found." });
     }
 
+    gym["__v"] = undefined;
```
Comment (RaduAlbastroiu) [inline Backend/api/gym/controller.js:17]: Not necessary you can remove this line

**Review by RaduAlbastroiu — APPROVED** [review]

**Pull request merged**